### ISSUE TITLE
DevTools: Console Object Inspector shows empty key

### ISSUE THREAD

**Issue by ArekSredzki** [atom/electron#4802]
* Electron version: 0.37.2
* Operating system: OS X

If you print an object to the console you are able to expand it.

A regression was added in 0.37.0 that causes a blank key (`""`) to show up.
This key is also expandable, if you try to do this it will cause the following error to appear:
```
DevTools was disconnected from the page.
Once page is reloaded, DevTools will automatically reconnect.
```

The following error will also occur when trying to expand the scope variables in the debugger.
For whatever reason, the empty key is "expanded", so when you expand the object containing the empty key, the error will occur.

**Comment by zcbenz** [comment]
Can you provide an example?

**Comment by ArekSredzki** [comment]
@zcbenz as in a picture?

**Comment by zcbenz** [comment]
I just did `console.log(require('electron'))` but didn't see what you described.

**Comment by deepak1556** [comment]
@zcbenz was able to reproduce with sending a simple object over ipc.

```
webContents.send('serialize', {test: "simple"})

ipcRenderer.on('serialize', function(e, msg) {
  console.log(Object.getOwnPropertyNames(msg))
})
```

returns an array with 2 elements instead of one. The cause seems to be use of `v8::Object::SetHiddenValue` ?

**Comment by deepak1556** [comment]
Couldnt find any related issue in upstream repos, so did a `git bisect` with the following code.

```c++
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include "src/base/logging.h"
#include "libplatform/libplatform.h"
#include "v8.h"

using namespace v8;

class ArrayBufferAllocator : public v8::ArrayBuffer::Allocator {
 public:
  virtual void* Allocate(size_t length) {
    void* data = AllocateUninitialized(length);
    return data == NULL ? data : memset(data, 0, length);
  }
  virtual void* AllocateUninitialized(size_t length) { return malloc(length); }
  virtual void Free(void* data, size_t) { free(data); }
};

int main(int argc, char* argv[]) {
  // Initialize V8.
  V8::InitializeICU();
  V8::InitializeExternalStartupData(argv[0]);
  Platform* platform = platform::CreateDefaultPlatform();
  V8::InitializePlatform(platform);
  V8::Initialize();

  // Create a new Isolate and make it the current one.
  ArrayBufferAllocator allocator;
  Isolate::CreateParams create_params;
  create_params.array_buffer_allocator = &allocator;
  Isolate* isolate = Isolate::New(create_params);
  {
    Isolate::Scope isolate_scope(isolate);
    HandleScope handle_scope(isolate);
    Local<Context> context = Context::New(isolate);
    Context::Scope context_scope(context);


    Local<Object> obj = Object::New(isolate);
    obj->SetHiddenValue(String::NewFromUtf8(isolate, "simple"),
                        Boolean::New(isolate, true));
    Local<Object> global = context->Global();
    Local<Value> global_object =
         global->Get(String::NewFromUtf8(isolate, "Object"));
    if (global_object->IsObject()) {
      Local<Value> method =
          global_object->ToObject()->Get(
              String::NewFromUtf8(isolate, "getOwnPropertyNames"));
      Local<Value> args[] = {obj};
      Local<Value> result =
          method.As<Function>()->Call(global_object, 1, args);
      if (result->IsArray()) {
        Local<Array> arr= result.As<Array>();
        CHECK(!arr->Length());
      }
    }
  }

  // Dispose the isolate and tear down V8.
  isolate->Dispose();
  V8::Dispose();
  V8::ShutdownPlatform();
  delete platform;
  return 0;
}
```

the hidden string started showing up since this [commit](https://github.com/v8/v8/commit/7d1263db477c812d40789c75be2f368e4c0b9769) and was fixed in v5.0.2 [here](https://github.com/v8/v8/commit/9eb4929502795b40324bd5ffda2840ec37793acd). Should we wait for the next chromium upgrade or use `v8::Object::SetPrivate` since `hiddenValue` is being deprecated anyway.

**Comment by ArekSredzki** [comment]
Thanks for looking into this further, you are correct that this occurs when sending objects on IPC.
Personally this is very interruptive in my development workflow so I would appreciate it being fixed asap, but could understand if you find the need to wait for the next chromium upgrade.
